### PR TITLE
[release-1.2] 🌱 bump dependencies

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Manual cherry-pick of https://github.com/kubernetes-sigs/cluster-api/pull/7701, https://github.com/kubernetes-sigs/cluster-api/pull/7689 https://github.com/kubernetes-sigs/cluster-api/pull/7695

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Needs a go mod tidy I think

**Comment by fabriziopandini** [comment]
not sure why github.com/MakeNowJust/heredoc v1.0.0  picks a different sha in CI than on my machines, need to investigate this a little bit

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         run: |
           make release-notes
       - name: Release
-        uses: softprops/action-gh-release@v1
+        uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844 # tag=v1@v1
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply .github/workflows/release.yml:34]: +1, @v1 looks like a duplication

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: .github/workflows/release.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply .github/workflows/release.yml]: fixed

**Comment by fabriziopandini** [comment]
(self-approving, it is a cherry-pick of dependabot work)
/approve

**Pull request merged**